File: 프로그래머스/코딩테스트 입문 캘린더/Day 21/120866. 안전지대/안전지대.py
def solution(board):
    around = [
        (-1, -1),
        (-1, 0),
        (-1, 1),
        (0, -1),
        (0, 0),
        (0, 1),
        (1, -1),
        (1, 0),
        (1, 1),
    ]
    count = 0
    n = len(board)
    check = [[True] * n for _ in range(n)]
    for i in range(n):
        for j in range(n):
            if board[i][j]:
                for p in around:
                    if 0 <= i + p[0] < n and 0 <= j + p[1] < n:
                        check[i + p[0]][j + p[1]] = False
    for i in range(n):
        for j in range(n):
            if check[i][j]:
                count += 1
    return count

File: 프로그래머스/lv0/120866. 안전지대/안전지대.js
function solution(board) {
  let n = board.length;
  const D = [
    [0, 1],
    [1, 0],
    [0, -1],
    [-1, 0],
    [1, 1],
    [-1, -1],
    [1, -1],
    [-1, 1],
  ];
  let dangerZone = new Set();
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      if (board[i][j] === 1) {
        dangerZone.add(`${i},${j}`);
        D.forEach(([dx, dy]) => {
          let x = i + dx;
          let y = j + dy;
          if (x >= 0 && x < n && y >= 0 && y < n) dangerZone.add(`${x},${y}`);
        });
      }
    }
  }
  console.log(dangerZone);
  return n ** 2 - dangerZone.size;
}